conftest.py

```diff
@@ -220,14 +215,16 @@ def _load_module(module_name, path):
 
 @pytest.fixture(autouse=True)
 def reset_config():
-    """Reset the configuration directory used in the tests (tests.constants.ROOT_FS) after every test
+    """Reset the configuration directory used in the tests (tests.constants.ROOT_FS) before every test
     """
     # Ideally this would just be constants.ROOT_FS.removetree("/"), but SQLAlchemy complains if the SQLite file is deleted.
+    # We have between the workers a shared constants.ROOT_DIR
     for e in constants.ROOT_FS.walk.files(exclude=["mscolab.db"]):
         constants.ROOT_FS.remove(e)
+    # mscolab_settings, mswms_settings, testdata is not changed so we can keep it
```

Comment:
When I keep it, like:

```
diff --git a/conftest.py b/conftest.py
index 9bf23264..08bd8560 100644
--- a/conftest.py
+++ b/conftest.py
@@ -221,10 +221,8 @@ def reset_config():
     # We have between the workers a shared constants.ROOT_DIR
     for e in constants.ROOT_FS.walk.files(exclude=["mscolab.db"]):
         constants.ROOT_FS.remove(e)
-    # mscolab_settings, mswms_settings, testdata is not changed so we can keep it
     for e in constants.ROOT_FS.walk.dirs(search="depth"):
-        if not e in ("/mscolab", '/mswms', '/mswms/testdata'):
-            constants.ROOT_FS.removedir(e)
+        constants.ROOT_FS.removedir(e)
     generate_initial_config()
     create_msui_settings_file("{}")
     read_config_file()
```

this gives

```
ERROR tests/_test_utils/test_multidict.py::TestCIMultiDict::test_multidict - fs.errors.ResourceNotFound: resource 'mswms_settings.py' not found
=================================== 8 skipped, 7 warnings, 548 errors in 24.08s ====================================
```
all have the same failure, they can't load mswms_settings.py

Reply:
I see. That error is originating from `generate_initial_config()`, so either your changes there broke it or they surfaced a bug that was already there.

Not cleaning the environment between tests is not a fix though.

I think the issue is that the reset fixture fully cleans the environment, but due to the way the constants are loaded (once, at the beginning of the test execution, which is a pain already because it makes it impossible to just use a different directory for each test, which would be the best solution) the mswms/ directory is created only once, when tests.constants is imported, and not recreated for the next test. I think this is a major architectural problem that needs to be re-thought, while the simple fix is to re-create the mswms/ directory at the beginning of `generate_initial_config`.